### DISCUSSION TITLE
Sourcemaps

### DISCUSSION THREAD

**Discussion by JeffreyBenjaminBrown** [gleam-lang/gleam#1341] [Ideas & suggestions]
Javascript supports [sourcemaps](https://www.html5rocks.com/en/tutorials/developertools/sourcemaps/) (sometimes as the two words "source maps"). These allow, at least for some languages that compile to Javascript, one to look up the source passage that compiled to the Javascript passage in question. This makes handy things like a REPL and a debugger feasible for the source language. (According to [Reddit user cstone949](https://www.reddit.com/r/erlang/comments/pgwq3f/is_there_a_language_like_erlang_with_a_stricter/hd903vt/?utm_source=reddit&utm_medium=web2x&context=3), sourcemaps are good for more than that,. but I don't know what. I've told him about this Github issue.)

Gleam compiles to Erlang. One could therefore, in theory, use the ordinary Erlang debugger to debug the compiled-to Erlang. At this time, however, one cannot use a debugger on source Gleam code.

(I don't know whether sourcemaps are the only way to make such things feasible for Gleam. I also don't know whether any language other than Javascript supports source maps.)

**Comment by lpil** [comment]
This is a great idea, one I think will bring a lot of value. I'm not familiar with how they work so I'm not sure how much work it would be, but let's definitely do it.

**Reply by pwtail** [reply]
> This is a great idea, one I think will bring a lot of value. I'm not familiar with how they work so I'm not sure how much work it would be, but let's definitely do it.

@lpil @JeffreyBenjaminBrown Is it? As far as I know, Erlang is not whitespace-based, so it's possible to have 1:1 source lines mapping. Or is it a goal to produce the most readable Erlang?

Are the line numbers in gleam code the same as in Erlang code currently?

**Reply by lpil** [reply]
There's no proposed additions to the Erlang target here. Nothing would change.

**Comment by dpchamps** [comment]
Hey @lpil, I'm interested in contributing this feature -- are you still open to it?

**Reply by lpil** [reply]
Yes please!
